examples/two-columns-layout/src/plugins/categoriesPlugin.tsx

```diff
@@ -0,0 +1,77 @@
+/** @jsx h */
+import {
+  AutocompletePlugin,
+  getAlgoliaResults,
+} from '@algolia/autocomplete-js';
+import { h } from 'preact';
+
+import { Breadcrumb } from '../components/Breadcrumb';
+import { ALGOLIA_PRODUCTS_INDEX_NAME } from '../constants';
+import { searchClient } from '../searchClient';
+import { CategoryHit } from '../types';
+
+export const categoriesPlugin: AutocompletePlugin<CategoryHit, {}> = {
```

Comment:
interesting that you chose to organise this as a plugin rather than a source, is that because you have the `if` in getSources which you prefer to be part of the "categories" block than inside the app?

Reply:
Yes it was a way to manage the empty/search state inside the plugin rather than in the app file
I don't know if we should stay with everything as a plugin or just keep `recentSearches` and `querySuggestions` as plugins and the rest as sources?